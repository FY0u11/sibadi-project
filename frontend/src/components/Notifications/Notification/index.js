import React, { useEffect } from 'react'

import timer from '../../../utils/timer'

import Notification from './Notification'

// delay before notifications are automatically closed (in milliseconds)
const NotificationContainer = ({ id, delay = 10000, closeNotification, ...rest }) => {
  useEffect(() => {
    timer(delay)(closeNotification, id)
  }, [])

  return <Notification {...rest} closeNotification={() => closeNotification(id)} />
}

export default NotificationContainer
