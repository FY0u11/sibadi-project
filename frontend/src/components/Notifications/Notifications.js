import React from 'react'

import './styles.css'
import Notification from './Notification'

const Notifications = ({ notifications, closeNotification }) => {
  return (
    <div className="my-notifications">
      {notifications.map(notification => {
        const { id, delay, level, title, message } = notification
        return (
          <Notification
            delay={delay}
            level={level}
            title={title}
            message={message}
            key={id}
            id={id}
            closeNotification={closeNotification}
          />
        )
      })}
    </div>
  )
}

export default Notifications
