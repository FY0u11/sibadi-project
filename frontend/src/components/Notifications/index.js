import React from 'react'
import { connect } from 'react-redux'

import { closeNotification } from '../../store/components/notifications/actions'

import Notifications from './Notifications'

const NotificationsContainer = ({ ...props }) => <Notifications {...props} />

const putStateToProps = state => ({ notifications: state.notifications.notifications })
const putActionsToProps = { closeNotification }

export default connect(putStateToProps, putActionsToProps)(NotificationsContainer)
