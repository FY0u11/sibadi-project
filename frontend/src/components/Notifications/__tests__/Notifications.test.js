import React from 'react'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import Notifications from '../Notifications'

afterEach(cleanup)

describe('Testing Notifications component', () => {
  it('It should be properly rendered', async () => {
    const notifications = [
      {
        id: 1,
        level: 'success',
        title: 'Test notification 1',
        message: 'Test message 1'
      },
      {
        id: 2,
        level: 'success',
        title: 'Test notification 2',
        message: 'Test message 2'
      }
    ]
    const { asFragment } = render(<Notifications notifications={notifications} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
