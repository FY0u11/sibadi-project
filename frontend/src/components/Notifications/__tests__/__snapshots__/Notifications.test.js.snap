// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Notifications component It should be properly rendered 1`] = `
<DocumentFragment>
  <div
    class="my-notifications"
  >
    <div
      class="card my-notification"
    >
      <h5
        class="card-header bg-success text-light"
      >
        Test notification 1
        <button
          class="ml-2 mb-1 close"
          type="button"
        >
          ×
        </button>
      </h5>
      <div
        class="card-body"
      >
        <p
          class="card-text"
        >
          Test message 1
        </p>
        <button
          class="btn btn-success text-light"
        >
          Close
        </button>
      </div>
    </div>
    <div
      class="card my-notification"
    >
      <h5
        class="card-header bg-success text-light"
      >
        Test notification 2
        <button
          class="ml-2 mb-1 close"
          type="button"
        >
          ×
        </button>
      </h5>
      <div
        class="card-body"
      >
        <p
          class="card-text"
        >
          Test message 2
        </p>
        <button
          class="btn btn-success text-light"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
